annotation-web/src/main/java/eu/europeana/annotation/web/service/controller/jsonld/BaseJsonldRest.java

```diff
@@ -91,10 +84,10 @@ protected ResponseEntity<String> storeAnnotation(MotivationTypes motivation, boo
 	    // 2. validate
 	    // check whether annotation with the given provider and identifier
 	    // already exist in the database
-	    if (annoId.getIdentifier() != null && getAnnotationService().existsInDb(annoId))
+	    if (getAnnotationService().existsInDb(annoIdentifier))
 		throw new ParamValidationI18NException(ParamValidationI18NException.MESSAGE_ANNOTATION_ID_EXISTS,
 			I18nConstants.ANNOTATION_VALIDATION,
-			new String[] { "/provider/identifier", annoId.toRelativeUri() });
+			new String[] { "identifier", String.valueOf(annoIdentifier) });
```

Comment:
the message should be changed to indicate with bad request when an id is provided to the create method